testnet/stacks-node/src/tests/signer.rs

```diff
@@ -329,3 +329,112 @@ fn test_stackerdb_dkg() {
     // Stop the coordinator
     assert!(running_coordinator.stop().is_none());
 }
+
+#[tracing_test::traced_test]
+#[test]
+fn signer_ping_mode_works() {
+    let num_signers: u32 = 2;
+    let num_keys: u32 = 2;
+    let publisher_private_key = StacksPrivateKey::new();
+    let signer_stacks_private_keys = (0..num_signers)
+        .map(|_| StacksPrivateKey::new())
+        .collect::<Vec<StacksPrivateKey>>();
+
+    let pox_contract_id =
+        QualifiedContractIdentifier::new(to_addr(&publisher_private_key).into(), "pox-4".into());
+    let stacker_db_contract_id = QualifiedContractIdentifier::new(
+        to_addr(&publisher_private_key).into(),
+        "hello-world".into(),
+    );
+
+    let (mut conf, _) = neon_integration_test_conf();
+
+    // Setup the signer and coordinator configurations
+    let signer_configs = build_signer_config_tomls(
+        &signer_stacks_private_keys,
+        num_keys,
+        &conf.node.rpc_bind,
+        &stacker_db_contract_id.to_string(),
+        Some(&pox_contract_id.to_string()),
+        Some(Duration::from_millis(128)), // Timeout defaults to 5 seconds. Let's override it to 128 milliseconds.
+    );
+
+    let mut running_signers = vec![];
+    // Spawn all the signers first to listen to the coordinator request for dkg
+    let mut signer_cmd_senders = Vec::new();
+    let mut signer_res_receivers = Vec::new();
+    for i in (0..num_signers).rev() {
+        let (cmd_send, cmd_recv) = channel();
+        let (res_send, res_recv) = channel();
+        let running_signer = spawn_signer(&signer_configs[i as usize], cmd_recv, res_send);
+        running_signers.push(running_signer);
+        cmd_send
+            .send(RunLoopCommand::Ping { payload_size: 0 })
+            .unwrap();
+        signer_cmd_senders.push(cmd_send);
+        signer_res_receivers.push(res_recv);
+    }
+
+    let coordinator_cmd_send = signer_cmd_senders.last().unwrap().clone();
+    let coordinator_res_recv = signer_res_receivers.last().unwrap();
+
+    // Let's scope the node to ensure stopping the signers doesn't cause issues.
+    {
+        // Build our simulated pox-4 stacks contract TODO: replace this with the real deal?
```

Comment:
Is this TODO resolvable in this PR?

Reply:
I'm carrying it over from `test_stackerdb_dkg.` I'll give it a try.

Reply:
I asked around, and the 'pox-4' file is nowhere to be found in `develop.` That may be a no atm.

Reply:
Correct; it's in `next`